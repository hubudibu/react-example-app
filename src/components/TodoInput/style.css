.TodoInput {
  width: 100%;
  box-sizing: border-box;
  padding: 1em 2em;
  border: none;
  border-bottom: 1px solid rgba(0,0,0,.2);
}
