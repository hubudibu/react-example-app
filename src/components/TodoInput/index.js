import React from 'react';
import './style.css';

class TodoInput extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      value: ''
    }
  }

  handleChange(text) {
    this.setState({
      value: text
    });
  }

  handleKeypress(keyCode) {
    if (keyCode === 13) {
      this.props.onSubmit(this.state.value);
      this.setState({
        value: ''
      })
    }
  }

  render() {
    return <input
              className="TodoInput"
              type="text"
              placeholder="Add new todo item..."
              value={this.state.value}
              onChange={event => this.handleChange(event.target.value)}
              onKeyPress={event => this.handleKeypress(event.charCode)}
              />;
  }
}

export default TodoInput;
