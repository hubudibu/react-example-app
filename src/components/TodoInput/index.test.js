import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import TodoInput from './';


it('TodoInput renders as in snapshot', () => {
  const wrapper = renderer.create(<TodoInput onSubmit={()=>{}} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('TodoInput value changes state', () => {
  const todoText = 'Sample todo';
  const input = mount(<TodoInput onSubmit={()=>{}} />);
  input.simulate('change', {target: {value: todoText}});
  expect(input.state()).toEqual({value: todoText});
});

it('TodoInput input changes value', () => {
  const todoText = 'Sample todo';
  const input = mount(<TodoInput onSubmit={()=>{}} />);
  input.simulate('change', {target: {value: todoText}});
  expect(input.find('input').get(0).value).toEqual(todoText);
});
