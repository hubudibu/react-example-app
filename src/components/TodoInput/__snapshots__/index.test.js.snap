exports[`test TodoInput renders as in snapshot 1`] = `
<input
  className="TodoInput"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Add new todo item..."
  type="text"
  value="" />
`;
